workers/securityworker/securityworker.py

```diff
@@ -28,9 +28,22 @@ def __init__(self):
 
         interval = app.config.get("SECURITY_SCANNER_INDEXING_INTERVAL", DEFAULT_INDEXING_INTERVAL)
         self.add_operation(self._index_in_scanner, interval)
+        self.add_operation(self._index_recent_manifests_in_scanner, interval)
 
     def _index_in_scanner(self):
-        self._next_token = self._model.index_manifests(self._next_token)
+        batch_size = self.app.config.get("SECURITY_SCANNER_V4_BATCH_SIZE", 0)
+        self._next_token = self._model.perform_indexing(self._next_token, batch_size)
+
+    def _index_recent_manifests_in_scanner(self):
+        batch_size = self.app.config.get("SECURITY_SCANNER_V4_BATCH_SIZE", 0)
+
+        try:
+            with GlobalLock(
+                "SECURITYWORKER_INDEX_RECENT_MANIFEST", lock_ttl=300, auto_renewal=True
+            ):
+                self._next_token = self._model.perform_indexing_recent_manifests(batch_size)
```

Comment:
is `self._next_token` being used in both `_index_in_scanner` and `_index_recent_manifests_in_sanner` would it cause a race condition?